include/sys/types.h

```diff
@@ -104,17 +105,6 @@
 
 #ifndef __ASSEMBLY__
 
-/* Floating point types */
```

Comment:
@xiaoxiang781216  I am running into problems with this change. It looks like we require now CONFIG_LIBM needs to be set to get the followinf removed from sys/types.h and moved to include/nuttx/lib/math.h

```
typedef float        float32;	
#ifndef CONFIG_HAVE_DOUBLE	
typedef float        double_t;	
typedef float        float64;	
#else	
typedef double       double_t;	
typedef double       float64;	
#endif	
```

But if we do set CONFIG_LIBM  we loose things like `fmaxf` from the toolchain headers that do not define the moved types .  Can you offer your motivation for this change and suggest how we are expected to use it.

Reply:
You don't need define CONFIG_LIBM to get these types, because math.h from the 3rd party will provide the similar typedef for you(e.g. newlib):
https://github.com/bminor/newlib/blob/master/newlib/libc/include/math.h#L151
Instead this patch avoid that NuttX typedef in sys/types.h is different from math.h in the 3rd party code which is important to keep the binary compatibility because all source code in NuttX use the same double_t/float_t typedef as 3rd party library.

Reply:
Yes but `float32` is not defined on the https://github.com/bminor/newlib/blob/master/newlib/libc/include/math.h#L151

Reply:
@xiaoxiang781216 the problem is when <cmath> includes <math.h> and cmath defines using ::float32 is is not defined.

Reply:
Yes, you are right. We should remove float32 and float64 which aren't defined by standard and all other POSIX like OS.